vms/rpcchainvm/ghttp/http_server.go

```diff
@@ -142,7 +141,21 @@ func (s *Server) Handle(ctx context.Context, req *httppb.HTTPRequest) (*emptypb.
 
 	s.handler.ServeHTTP(writer, request)
 
-	return &emptypb.Empty{}, clientConn.Close()
+	if err := clientConn.Close(); err != nil {
+		return nil, fmt.Errorf("failed to close client conn: %w", err)
+	}
+
+	header := make([]*httppb.Element, len(writerHeaders))
+	for k, v := range writerHeaders {
+		header = append(header, &httppb.Element{
+			Key:    k,
+			Values: v,
+		})
+	}
```

Comment:
I believe this will result in `header` having a final length of `2 * len(writerHeaders)` where the first `len(writerHeaders)` entries are `nil`.
```suggestion
	header := make([]*httppb.Element, 0, len(writerHeaders))
	for k, v := range writerHeaders {
		header = append(header, &httppb.Element{
			Key:    k,
			Values: v,
		})
	}
```

Reply:
This doesn't panic, rather it ends up adding a `"": nil` entry to the headers.